lib/index.js

```diff
@@ -11,22 +11,32 @@ module.exports = (server, options, next) => {
 
     server.decorate('reply', 'push', function (response, path, headers) {
 
-        if (typeof response === 'string') {
+        if (typeof response === 'string' || Array.isArray(response)) {
             headers = path;
             path = response;
             response = this.request.response;
         }
 
         Hoek.assert(response && !response.isBoom, 'Must server-push with a non-error response.');
+        Hoek.assert(!!path, 'Must provide a path to the resource to be pushed.');
+
+        // now we force path to be an Array (makes life easier later)
+        if (!Array.isArray(path)) {
+            path = [path];
+        }
 
         headers = headers || {};
 
-        // Adjust for relative path
-        path = (path[0] === '/') ? path : `${this.realm.modifiers.route.prefix || ''}/${path}`;
+        const pushes = [];
+        for (const currentPath of path) {
+            // Adjust for relative path
+            currentPath = (currentPath[0] === '/') ? currentPath : `${this.realm.modifiers.route.prefix || ''}/${currentPath}`;
```

Comment:
The issue was this assignment to `currentPath`, which is a constant.  I'm working on this today, so I'm going to go ahead and fix it!